Pipeline to provision OSD on AWS

trepel opened the pull request:
Adds pipeline for OSD on AWS cluster creation.

`oc apply -k infra/ -n desired-ns`

One has to have secrets created in desired namespace:

`kubectl create secret generic kua-ocm-stage-client-credentials --from-literal=CLIENT_ID="xxx" --from-literal=CLIENT_SECRET="xxx" -n desired-ns`

`kubectl create secret generic kua-aws-credentials --from-literal=AWS_ACCOUNT_ID="xxx" --from-literal=AWS_ACCESS_KEY_ID="xxx" --from-literal=AWS_SECRET_ACCESS_KEY="xxx" -n desired-ns`

trepel commented:
@averevki some things came to my mind:
1) Instead of installing the OCM we can maintain our custom image where it is already installed. I actually think that it is better approach.

2) It should be possible to extract `ocm login...` into separate Task by storing ocm.json into shared workspace and use it via OCM_CONFIG env var. It would allow to reuse the Task in other similar pipelines (ROSA, OSD on GCP, ARO provisioning)

3) In general I would like to split that script as much as possible. I would also separate that "waiting for cluster to become ready" into separate Task. It should be doable via creating a result (with e.g. cluster ID) so subsequent Tasks can consume it as input param and do that waiting. Again that would allow to re-use the Task in other pipelines

4) I should probably specify some requests/limits in `computeResources` for the Tasks

trepel commented:
@averevki  I addressed my four bulletpoints from the comment above in https://github.com/trepel/testsuite-pipelines/commits/osd-provisioning-split/. Feel free to take a look.

averevki commented:
@trepel very nice job! Maybe we can extend the @azgabur tooling image to also include the OCM, we need to ask him where he stores it. Also, I would expand the project readme to include all the prerequisites for this pipeline run and how to install it.

Otherwise, I'm ready to merge the version you sent in the commit with task splitting and we can try it in action

trepel commented:
@azgabur I tidied this up a bit, please review once you can spare some time.
- I am running the pipeline atm in `trepel` ns in kua-multi-1 so you can take a look there.
- Custom image is used, I will unify the images in use across the pipelines in a separate PR

trepel commented:
@azgabur I have just realized that it would be good to print/store as results cluster information (at least kubeadmin password, api url, console url. I will look into it and update the PR

trepel commented:
@azgabur I added a Task to show/return cluster details (including credentials). I also added a pipeline to delete the cluster - given the fact that we are not members of the same HCC org this is important - anyone can delete the cluster via pipeline.

I think it is ready for review now. You can take a look at the trepel ns in kua-multi-1 to see it working. Look at 'test' pipeline run to see that get-cluster-credentials Task in action.

trepel commented:
@azgabur I added another commit with pipeline for OSD on GCP. A lot of re-using of Tasks so it turned out to be quite easy to add.

azgabur commented on infra/osd-aws-pipeline.yaml:
```diff
@@ -0,0 +1,78 @@
+spec:
+  description: Provision OSD on AWS
+  params:
+    - default: kua-aws-credentials
+      description: Name of secret that contains credentials for osdCcsAdmin
+      name: aws-credentials
+      type: string
+    - default: kua-ocm-stage-client-credentials
+      description: Name of secret that contains client ID and client secret to log into OCM org
+      name: ocm-credentials
+      type: string
```
I am thinking if it would not be better to use Secret object to store credentials like this. Firstly we could save credentials once in cluster and when running the pipeline we would just choose the cluster name and staging/prod env. Secondly the credentials would not be stored in PipelineRun mostly in plaintext - not a huge problem but maybe better practice.
